change-validation/src/main/scala/com/normation/plugins/changevalidation/ChangeRequestJson.scala

```diff
@@ -0,0 +1,961 @@
+      rule: JRRule
+  ) extends RuleChangeJson
+
+  final case class RuleDeleteChangeJson(
+      rule: JRRule
+  ) extends RuleChangeJson
+
+  final case class RuleModifyChangeJson(
+      change: ModifyRuleJson
+  ) extends RuleChangeJson
+
+  final case class ModifyRuleJson(
+      id:                                                 RuleId,
+      @jsonField("displayName") modName:                  SimpleDiffOrValueJson[String],
+      @jsonField("shortDescription") modShortDescription: SimpleDiffOrValueJson[String],
+      @jsonField("longDescription") modLongDescription:   SimpleDiffOrValueJson[String],
+      @jsonField("directives") modDirectiveIds:           SimpleDiffOrValueJson[Set[DirectiveId]],
+      @jsonField("targets") modTarget:                    SimpleDiffOrValueJson[Set[RuleTarget]],
+      @jsonField("enabled") modIsActivatedStatus:         SimpleDiffOrValueJson[Boolean]
+  )
+
+  object ModifyRuleJson {
+    implicit lazy val ruleIdEncoder:      JsonEncoder[RuleId]         = JsonEncoder[String].contramap[RuleId](_.serialize)
+    implicit lazy val directiveIdEncoder: JsonEncoder[DirectiveId]    = JsonEncoder[String].contramap[DirectiveId](_.serialize)
+    implicit lazy val ruleTargetEncoder:  JsonEncoder[RuleTarget]     = JsonEncoder[String].contramap[RuleTarget](_.target)
+    implicit lazy val encoder:            JsonEncoder[ModifyRuleJson] =
+      DeriveJsonEncoder.gen[ModifyRuleJson]
+
+    def from(
+        modifyRuleDiff: ModifyRuleDiff,
+        initialState:   Rule
+    ): ModifyRuleJson = {
+      ModifyRuleJson(
+        modifyRuleDiff.id,
+        SimpleDiffOrValueJson.withDefault(modifyRuleDiff.modName, initialState.name),
+        SimpleDiffOrValueJson.withDefault(modifyRuleDiff.modShortDescription, initialState.shortDescription),
+        SimpleDiffOrValueJson.withDefault(modifyRuleDiff.modLongDescription, initialState.longDescription),
+        SimpleDiffOrValueJson.withDefault(modifyRuleDiff.modDirectiveIds, initialState.directiveIds),
+        SimpleDiffOrValueJson.withDefault(modifyRuleDiff.modTarget, initialState.targets),
+        SimpleDiffOrValueJson.withDefault(modifyRuleDiff.modIsActivatedStatus, initialState.isEnabledStatus)
+      )
+    }
+  }
+  implicit val jrRuleTransformer: Transformer[Rule, JRRule] = Transformer
+    .define[Rule, JRRule]
+    .enableBeanGetters
+    .withFieldComputed(_.id, _.id.serialize)
+    .withFieldRenamed(_.name, _.displayName)
+    .withFieldComputed(_.categoryId, _.categoryId.value)
+    .withFieldComputed(_.directives, _.directiveIds.map(_.serialize).toList.sorted)
+    .withFieldComputed(_.targets, _.targets.toList.sortBy(_.target).map(t => JRRuleTarget(t)))
+    .withFieldRenamed(_.isEnabledStatus, _.enabled)
+    .withFieldComputed(_.tags, x => JRTags.fromTags(x.tags))
+    .withFieldConst(_.policyMode, None)      // not needed
+    .withFieldConst(_.status, None)          // not needed
+    .withFieldConst(_.changeRequestId, None) // not needed
```

Comment:
I think they should. If they are not in 7.3/8.0, it's a bug - can you check and do what is needed accordingly ?

Reply:
Here specifically these fields are used when there is a "new" Rule change request or a "deletion" of a Rule change request (`JRRule` is only used for serializing the resulting CR in both cases). 

The previous lift-json code replaced by this one does not in fact serialize these fields : 
```
  def serializeRule(rule: Rule, crId: Option[ChangeRequestId]): JValue = {
    (("changeRequestId"   -> crId.map(_.value.toString))
    ~ ("id"               -> rule.id.uid.value)
    ~ ("revision"         -> rule.id.rev.toOptionString)
    ~ ("displayName"      -> rule.name)
    ~ ("categoryId"       -> rule.categoryId.value)
    ~ ("shortDescription" -> rule.shortDescription)
    ~ ("longDescription"  -> rule.longDescription)
    ~ ("directives"       -> rule.directiveIds.toList.map(_.serialize).sorted)
    ~ ("targets"          -> rule.targets.map(_.toJson))
    ~ ("enabled"          -> rule.isEnabledStatus)
    ~ ("system"           -> rule.isSystem)
    ~ ("tags"             -> serializeTags(rule.tags)))
  }
```

A way to avoid this confusion is the create another case class like `JRRule` but with only those serialized fields, but I chose to reuse `JRRule`, it has all we need (even more)